fsw/src/crypto.c

```diff
@@ -3134,220 +3259,119 @@ int32 Crypto_TC_ProcessSecurity( char* ingest, int* len_ingest,TC_t* tc_sdls_pro
-            #ifdef TC_DEBUG
-                OS_printf("\t iv[%d] = 0x%02x\n", x-byte_idx, tc_sdls_processed_frame->tc_sec_header.iv[x-byte_idx]);
-            #endif
+            case SA_PLAINTEXT:
+                OS_printf(KBLU "Processing a TC - CLEAR!\n" RESET);
+                break;
+            case SA_AUTHENTICATION:
+                OS_printf(KBLU "Processing a TC - AUTHENTICATED!\n" RESET);
+                break;
+            case SA_ENCRYPTION:
+                OS_printf(KBLU "Processing a TC - ENCRYPTED!\n" RESET);
+                break;
+            case SA_AUTHENTICATED_ENCRYPTION:
+                OS_printf(KBLU "Processing a TC - AUTHENTICATED ENCRYPTION!\n" RESET);
+                break;
         }
-        byte_idx += sa_ptr->shivf_len;
-        report.snval = tc_sdls_processed_frame->tc_sec_header.iv[sa_ptr->shivf_len-1];
+    #endif
 
-        #ifdef DEBUG
-            OS_printf("\t tc_sec_header.iv[%d] = 0x%02x \n", sa_ptr->shivf_len-1, tc_sdls_processed_frame->tc_sec_header.iv[sa_ptr->shivf_len-1]);
-            OS_printf("\t sa[%d].iv[%d] = 0x%02x \n", tc_sdls_processed_frame->tc_sec_header.spi, sa_ptr->shivf_len-1, sa_ptr->iv[sa_ptr->shivf_len-1]);
-        #endif
+    // TODO: Calculate lengths when needed
+    uint8 fecf_len = FECF_SIZE;
+    if(current_managed_parameters->has_fecf==TC_NO_FECF) { fecf_len = 0; }
 
-        // Check IV is in ARCW
-        if ( crypto_config->ignore_anti_replay==TC_IGNORE_ANTI_REPLAY_FALSE )
-        {
+    uint8 segment_hdr_len = SEGMENT_HDR_SIZE;
+    if(current_managed_parameters->has_segmentation_hdr==TC_NO_SEGMENT_HDRS) { segment_hdr_len = 0; }
 
-            if ( Crypto_window(tc_sdls_processed_frame->tc_sec_header.iv, sa_ptr->iv, sa_ptr->shivf_len,
-                               sa_ptr->arcw[sa_ptr->arcw_len-1]) != CRYPTO_LIB_SUCCESS )
-            {
-                report.af = 1;
-                report.bsnf = 1;
-                if (log_summary.rs > 0)
-                {
-                    Crypto_increment((uint8*)&log_summary.num_se, 4);
-                    log_summary.rs--;
-                    log.blk[log_count].emt = IV_WINDOW_ERR_EID;
-                    log.blk[log_count].emv[0] = 0x4E;
-                    log.blk[log_count].emv[1] = 0x41;
-                    log.blk[log_count].emv[2] = 0x53;
-                    log.blk[log_count].emv[3] = 0x41;
-                    log.blk[log_count++].em_len = 4;
-                }
-                OS_printf(KRED "Error: IV not in window! \n" RESET);
-                #ifdef OCF_DEBUG
-                Crypto_fsrPrint(&report);
-                #endif
-                status = OS_ERROR;
-            }
-            else
+    // Check FECF
+    if(current_managed_parameters->has_fecf==TC_HAS_FECF)
+    {
+        if(crypto_config->crypto_check_fecf == TC_CHECK_FECF_TRUE)
+        {
+            uint16 received_fecf = (tc_sdls_processed_frame->tc_header.fl-1 & 0xFF00) | (tc_sdls_processed_frame->tc_header.fl & 0x00FF); 
```

Comment:
This logic is incorrect.  It should be as follows:
uint16 received_fecf = (((ingest[tc_sdls_processed_frame->tc_header.fl - 1] << 8) & 0xFF00) | (ingest[tc_sdls_processed_frame->tc_header.fl] & 0x00FF));